infra-pipeline/pipeline/inputs.tf

```diff
@@ -49,6 +49,25 @@ variable "tf_root" {
   description = "The location (source repo and path) to run terraform commands"
 }
 
+variable "notification_topics" {
```

Comment:
wondering if it makes sense to rename this variable- something like `group_emails`. 

Additionally, we've used a pattern in the past, where we create create a parent object and nest the related nodes within. In this case, we may be able to refactor/add `notification_rules` into a notification object like:

```
variable "notifications" {
  group_emails = map(object({emails = set(string)}))
  rules = type = list(object({...
}
```

Reply:
@isaac-looker I originally named it "notification_groups", but changed it to "notification_topics" to prevent any confusion around what it actually represents (SNS Topics).  I wanted to avoid anything too specific, because technically it could also include phone numbers, HTTP(S) endpoints, Lambda functions, etc.  In fact, I would be a little disappointed if they _didn't_ hook this into Slack at some point.  I don't object to changing it to something else, but I do want to keep it generic enough that it won't become confusing when it gets extended later.

I kept the rules separate from the groups because I was planning on adding per-action notifications.  With the pipeline itself, you can't distinguish between stages/actions for notification triggers; it's all or nothing.  By implementing notification groups independently from rules, it becomes much easier to add triggers to CodeBuild status changes that can be defined per-action rather than per-pipeline.  Maybe the security team doesn't want to be notified every time the pipeline fails, but they do want to know when the code scan fails.  The dev team may want to know if the dev pipeline fails to build, but the infra team may only care if it's during the deployment action.  Since the pipeline spans multiple environments, you may have cases where one group only cares about the "prod" environment and not the "dev" or "staging" environments.

The notification topics are also used for approvals, so if you require someone to sign off prior to deployment, you can add a block like this:

```
approval = {
  required = true
  topic = "approvers"
}
```